### PR TITLE
[buildkite] Migrate DRA workflows

### PR DESCRIPTION
**Pull request by brianseeders**
- Bring over DRA-related scripts from Jenkins, and adapt for Buildkite
- Create dra-update-staging pipeline that checks all required branches for dependency updates in a single build (rather than one build per branch)
- Create dra-workflow trigger pipeline that triggers DRA workflow for staging/snapshot for each active branch
- Create dra-workflow pipeline that builds and publishes artifacts
- One script is now used for both staging and snapshot (license key configuration only applied in the case of staging, as before)

The Buildkite pipelines for this already exist, we just have to uncomment the schedules to effectively "deploy" them.

Builds:
- [Trigger workflows](https://buildkite.com/elastic/elasticsearch-dra-trigger-workflows/builds/3) - okay that triggered builds failed, the pipelines don't exist yet on those branches
- [Update staging](https://buildkite.com/elastic/elasticsearch-dra-update-staging/builds/1#018a51dd-01c8-423b-b27a-95add8f3b412)
- [dra workflow](https://buildkite.com/elastic/bseeders-elasticsearch-dra/builds/9#018a51b5-5e56-4735-ae4c-d41d75ae5e60) - for this example build, the release manager command just echoes at the end, instead of actually publishing anything

### CODE REVIEW

File: .buildkite/pipelines/dra-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+steps:
```
Comment (mark-vieira) [inline .buildkite/pipelines/dra-workflow.yml:1]: Are we defining this in our own terraform for now?

File: .buildkite/scripts/dra-workflow.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+BEATS_BUILD_ID="$(./.ci/scripts/resolve-dra-manifest.sh beats "$RM_BRANCH" "$ES_VERSION" "$WORKFLOW")"
+ML_CPP_BUILD_ID="$(./.ci/scripts/resolve-dra-manifest.sh ml-cpp "$RM_BRANCH" "$ES_VERSION" "$WORKFLOW")"
+
+LICENSE_KEY_ARG=""
+BUILD_SNAPSHOT_ARG=""
+
+if [[ "$WORKFLOW" == "staging" ]]; then
+  LICENSE_KEY=$(mktemp -d)/license.key
+  # Intentionally leaving this secret here instead of a hook
+  # Since it's getting redirected straight to a file, it's probably even safer to just leave it here rather than stick it in an env var for a short period of time
```
Comment (mark-vieira) [inline .buildkite/scripts/dra-workflow.sh:40]: Also, this isn't really a "secret". It's the public key used for license validation so it's not actually sensitive.

File: .buildkite/scripts/dra-workflow.trigger.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (mark-vieira) [inline .buildkite/scripts/dra-workflow.trigger.sh:1]: I'm not seeing what's actually referencing this script. Is it another external pipeline definition?

**Review by mark-vieira — APPROVED** [review]

File: .buildkite/scripts/dra-workflow.sh (hunk 2 above)
Comment (brianseeders) [reply .buildkite/scripts/dra-workflow.sh]: Ah, I should have paid closer attention that it was just the public key, thanks. Changed comment about this

**Pull request merged**